src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetsControllerTests.cs

```diff
@@ -1025,30 +1072,38 @@ public async Task PublicationDoesNotExist_Returns404()
                 .WithFile(file)
                 .WithReleaseVersion(releaseVersion);
 
-            await TestApp.AddTestData<ContentDbContext>(context => context.ReleaseFiles.Add(releaseFile));
+            await fixture.GetContentDbContext().AddTestData(context => context.ReleaseFiles.Add(releaseFile));
 
             DataSet dataSet = DataFixture.DefaultDataSet().WithStatusPublished().WithPublicationId(Guid.NewGuid());
 
-            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+            await fixture.GetPublicDataDbContext().AddTestData(context => context.DataSets.Add(dataSet));
 
             var response = await GetDataSet(dataSet.Id);
 
             response.AssertNotFound();
         }
 
-        private async Task<HttpResponseMessage> GetDataSet(Guid dataSetId, HttpClient? client = null)
+        private async Task<HttpResponseMessage> GetDataSet(Guid dataSetId, ClaimsPrincipal? user = null)
         {
-            var user = DataFixture.AuthenticatedUser().WithClaim(SecurityClaimTypes.AccessAllPublications.ToString());
+            // TODO EES-6450 - are we happy with this approach for handling default or specific users?
+            // Are we OK always having to do an explicit "RegisterTestUser" or should we set a user on
+            // Fixture and then CreateClient() always uses that user until changed?  Should be safe but
+            // is more "magic".
+            var defaultUser = DataFixture
+                .AuthenticatedUser()
+                .WithClaim(SecurityClaimTypes.AccessAllPublications.ToString());
 
-            client ??= TestApp.SetUser(user).CreateClient();
+            fixture.RegisterTestUser(user ?? defaultUser);
+
+            var client = fixture.CreateClient().WithUser(user ?? defaultUser);
```

Comment:
I was going to leave a similar comment on a previous test class.

Is it not possible to set the user once on the fixture which will register it if necessary and then `CreateClient` pick it up?

It would remove a bit of duplication, but like you say it's a bit more magic so others might not like it. It feels like there's less chance for things to get missed though if it was a bit more automated.